src/library/searchquery.cpp

```diff
@@ -166,6 +172,46 @@ QString TextFilterNode::toSql() const {
     return concatSqlClauses(searchClauses, "OR");
 }
 
+CrateFilterNode::CrateFilterNode(const QSqlDatabase& database,
+                                 const QString& argument)
+    : m_database(database),
+      m_argument(argument) {
+    getTrackIds();
+}
+
+void CrateFilterNode::getTrackIds() {
+    FieldEscaper escaper(m_database);
+    QString escapedArgument = escaper.escapeString(kSqlLikeMatchAll + m_argument + kSqlLikeMatchAll);
+
+    QSqlQuery query(m_database);
```

Comment:
Nope, toSql() and match() serve a different purpose. toSql() is the main query for the accessing the database while match() is only applied for post-filtering all tracks that are already cached in memory, e.g. all dirty tracks that have been modified recently. Without any dirty tracks match() will not be invoked!

You can circumvent the const-correctness by using mutable members behind the scenes. But we can add this optimization later and avoid those dirty tricks ;)